addon-test-support/setup-global-a11y-hooks.ts

```diff
@@ -0,0 +1,38 @@
+import { _registerHook } from '@ember/test-helpers';
+import { InvocationStrategy } from './invocation-strategies';
+import { getRunOptions } from './run-options';
+import { HookUnregister } from '@ember/test-helpers';
+import a11yAudit from './audit';
+
+let _unregisterHooks: HookUnregister[] = [];
+
+/**
+ *
+ * @param shouldAudit Invocation strategy function that determines whether to run the audit helper or not.
+ * @param audit Optional audit function used to run the audit. Allows for providing either a11yAudit, a11yAuditIf,
+ *              or custom audit implementation.
+ */
+export function setupGlobalA11yHooks(
+  shouldAudit: InvocationStrategy,
+  audit: (...args: any[]) => PromiseLike<void> = a11yAudit
+) {
+  ['visit', 'click'].forEach((helperName) => {
```

Comment:
This is just a general thought and not related to the code at hand. Would the host app be able to unregister a hook, if they choose to do so? I'm thinking that perhaps a particular test does some setup work after a `visit`, making the `a11yAudit` not particularly useful at that point in time.

Reply:
Can you think of a specific use case we would need to support? I'd be helpful to have a specific use case to work from so we can determine the best API.

Reply:
Modals and similar types of overlays are common examples, which require a set of user actions to render on screen. We would essentially end up with some overlap if a prior test covers the main page. However, I don't think this is something we should tackle at this point, as we're going to run into cases of overlap, regardless.

Reply:
What are the drawbacks of having overlaps if that results in less overhead for the engineer writing the tests? Is it a performance concern with reading DOM?

Reply:
One is performance, the other is mitigating overly noisy output when it comes to a11y violations. However, the latter can likely be addressed via deduping violation output.